### PR TITLE
Remove help button from debug window

### LINKED ISSUE #547: Clicking on the Help Button does nothing
If you click on the Help Button in the Debug Console window, the Mouse Cursor changes to a "forbidden" sign, but no Help Menu is opened.
I propose to either remove the Button or at least return a Work in Progress message.

### PR DESCRIPTION
**Pull request by chrstphrchvz**
Closes #547

Adapted from [this SO answer](https://stackoverflow.com/a/30934930/4896937), but I don't know if this is the right way to do so.

I have **not** tested this commit since I do not have Windows build environment set up. Can anyone volunteer to test it?

**Edit:** @barton2526 has offered to test this

### CODE REVIEW

**Review by iFoggz — APPROVED** [review]
Approved pending @barton2526 windows test

**Comment by barton2526** [comment]
Test shows no change in Windows GUI 😦

**Comment by barton2526** [comment]
Test shows no change in Windows GUI 😦

**Comment by barton2526** [comment]
Test shows no change in Windows GUI 😦 

issueunresolved

**Comment by barton2526** [comment]
Test shows no change in Windows GUI 😦 

issueunresolved

**Comment by barton2526** [comment]
Test shows no change in Windows GUI 😦 

issueunresolved

**Comment by barton2526** [comment]
Test shows no change in Windows GUI 😦 

issueunresolved

**Comment by barton2526** [comment]
Test shows no change in Windows GUI 😦 

issueunresolved

**Comment by barton2526** [comment]
Test shows no change in Windows GUI 😦 

issueunresolved

**Comment by barton2526** [comment]
Test shows no change in Windows GUI 😦 

issueunresolved

**Comment by chrstphrchvz** [comment]
This attempt really was a guess, I wasn't sure at all it was going to work since I'm not familiar with these specific parts of Qt or C++, so maybe to someone with more C++ Qt knowledge they might have a better idea (@barton2526 suggested tagging @skcin). The answers I had found online for turning off the help button tend to be using the `setWindowFlags()` setter on a specific window then using `show()`, whereas I haven't seen any examples doing this from a subclass. Nor have I even seen any examples that set a subclass' constructor's `Qt::WindowFlags f` parameter's default values to anything other than `0` or `Qt::WindowFlags()`.

**Comment by chrstphrchvz** [comment]
Not sure if any similar wallet programs have already found a solution for this, the only ones I use are for Gridcoin and Curecoin. The Bitcoin core wallet has since [switched away from using QDialog](https://github.com/bitcoin/bitcoin/commit/4a8fc152a957e54e6dd910de4382678f5c405198) for the debug window, but that looks like it would be a more involved change for Gridcoin to similarly adopt.

**Comment by denravonska** [comment]
What about this method? https://stackoverflow.com/a/49905954

> QApplication::setAttribute(Qt::AA_DisableWindowContextHelpButton)

Edit: nvm, that's globally.

**Comment by iFoggz** [comment]
The help (Qt::WindowContextHelpButtonHint) is default placed if I remember correctly. However if you specify the window flags it will not be added as you specify the windows conditions. I believe what should of been done was something like this:

RPCConsole::RPCConsole(QWidget *parent) :
    QDialog(parent, Qt::WindowSystemMenuHint | Qt::WindowTitleHint),
    ui(new Ui::RPCConsole),
    historyPtr(0)


If you were to add the "Qt::WindowContextHelpButtonHint" then it would show again. 

edit:

Figure out which window flags we need and set it accordingly

**Comment by jamescowens** [comment]
@iFoggz You want to pick this up?

**Comment by iFoggz** [comment]
if i have someone to compile the windows exe and test it since i have no windows machines in my house lol

File: src/qt/rpcconsole.cpp [hunk 1]
```diff
@@ -206,7 +206,7 @@ void RPCExecutor::request(const QString &command)
     }
 }
 
-RPCConsole::RPCConsole(QWidget *parent) :
+RPCConsole::RPCConsole(QWidget *parent, Qt::WindowFlags f) :
     QDialog(parent),
```
Comment (iFoggz) [inline src/qt/rpcconsole.cpp:210]: I tested with @barton26 help a similar test with https://github.com/iFoggz/Gridcoin-Research/commit/39f10878d16f4522be666c58fcd7d2acd537e6b9 . The issue I see here is `QDialog(parent),` whould be `QDialog(parent, f),` this should make this work or u can just do what the commit shows instead.

**Pull request closed without merging**